lib/logger.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+import logging.handlers
+import threading
+import queue
+
+# Set log level from environment variable
+log_level = os.getenv('YB_DRIVER_LOG_LEVEL', 'WARNING').upper()
+
+# Mapping of string level names to logging constants
+level_mapping = {
+    'DEBUG': logging.DEBUG,
+    'INFO': logging.INFO,
+    'WARNING': logging.WARNING,
+    'ERROR': logging.ERROR,
+    'CRITICAL': logging.CRITICAL
+}
+
+# Default to 'WARNING' level if the environment variable is invalid
+log_level = level_mapping.get(log_level, logging.WARNING)
+
+# Create a queue for logging
+log_queue = queue.Queue()
+
+# Create a handler that writes log messages to the queue
+queue_handler = logging.handlers.QueueHandler(log_queue)
+
+# Create a stream handler for console output
+console_handler = logging.StreamHandler()
+
+# Set up logging with queue handler and console handler
+logging.basicConfig(
+    level=log_level,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s - [%(filename)s:%(lineno)d]',
+    handlers=[queue_handler]  # Use queue handler for thread safety
+)
+
+# Create a logger instance
+logger = logging.getLogger(__name__)
+
+# Function that listens for logs from the queue and writes them
+def log_listener():
+    while True:
+        try:
+            # Get log record from the queue
+            record = log_queue.get()
+            if record is None:
+                break  # Stop listener
```

Comment:
If the listener_thread is exiting upon an empty queue, how will subsequent log messages get printed?

Reply:
The listener_thread won't exit on empty queue, If the queue is empty, the get() method will block until a new log record is added to the queue.
The listener_thread would exit it t retrieves a `None` value from the queue or will be terminated by the interpreter once the application finishes.